tests/test_scan.py

```diff
@@ -367,6 +367,52 @@ def test_scan_execute_with_soda_server_client_scan_result_does_not_contain_any_e
     assert not scan_result.has_errors()
 
 
+def test_test_results_to_data_frame(spark_session: SparkSession) -> None:
+    """Test conversions of test_result to dataframe."""
+    expected = spark_session.createDataFrame(
+        [
+            {
+                "test": {
+                    "id": "id",
+                    "title": "title",
+                    "expression": "expression",
+                    "metrics": ["metrics"],
+                    "column": "column",
+                    "expression_delimiters": ["<"],
+                    "source": "soda-sql",
+                },
+                "passed": True,
+                "skipped": False,
+                "values": {"value": 10},
+                "error": Exception("exception"),
```

Comment:
We need to think about what we do with this error. In the schema you have, it is converted to a `StringType`, that makes sense to me even tough it is not the _actual_ type mentioned in the data class. If we go for `StringType` than change `Exception("exception")` to `"exception"